packages/@cosmos-building-blocks/service/src/ecs/ecs-service.ts

```diff
@@ -58,92 +59,98 @@ export class EcsService extends Construct {
       vpc,
       cluster,
       httpListener,
-      containerProps,
+      logProps,
       taskProps,
+      containerProps,
       serviceProps,
       targetGroupProps,
       routingProps,
       scalingProps,
-      logProps,
     } = props;
 
-    this.LogGroup = new LogGroup(this, 'Logs', logProps);
+    this.logGroup = new LogGroup(this, 'Logs', {
+      retention: RetentionDays.ONE_MONTH,
+      ...logProps,
+    });
 
-    this.TaskDefinition = new Ec2TaskDefinition(this, 'Task', {
+    this.taskDefinition = new Ec2TaskDefinition(this, 'Task', {
       ...taskProps,
     });
 
-    this.Container = this.TaskDefinition.addContainer('Container', {
-      memoryLimitMiB: 256,
-      logging: LogDrivers.awsLogs({ logGroup: this.LogGroup, streamPrefix: `Container` }),
+    this.container = this.taskDefinition.addContainer('Container', {
+      memoryReservationMiB: 256,
+      logging: LogDrivers.awsLogs({ logGroup: this.logGroup, streamPrefix: `Container` }),
       ...containerProps,
     });
 
-    this.Container.addPortMappings({
+    this.container.addPortMappings({
       containerPort: 80,
       protocol: Protocol.TCP,
       ...containerProps.port,
     });
 
-    this.Service = new Ec2Service(this, 'Service', {
+    this.service = new Ec2Service(this, 'Service', {
       desiredCount: 1,
       placementStrategies: [PlacementStrategy.spreadAcross(BuiltInAttributes.AVAILABILITY_ZONE)],
       ...serviceProps,
-      taskDefinition: this.TaskDefinition,
+      taskDefinition: this.taskDefinition,
       cluster: cluster,
     });
 
     if (routingProps) {
       if (!httpListener) throw new Error('To enable routing, Http Listener is required');
 
-      this.TargetGroup = new ApplicationTargetGroup(this, 'ServiceTargetGroup', {
+      this.targetGroup = new ApplicationTargetGroup(this, 'ServiceTargetGroup', {
         protocol: ApplicationProtocol.HTTP,
         deregistrationDelay: Duration.seconds(0),
+        healthCheck: {
+          path: '/health',
+        },
         ...targetGroupProps,
         vpc: vpc,
         targets: [
-          this.Service.loadBalancerTarget({
+          this.service.loadBalancerTarget({
             containerName: 'Container',
           }),
         ],
       });
 
-      this.ListenerRule = new ApplicationListenerRule(this, 'ServiceRule', {
-        priority: getRoutingPriority(routingProps),
+      this.listenerRule = new ApplicationListenerRule(this, 'ServiceRule', {
+        priority: getRoutingPriorityFromListenerProps(routingProps),
         ...routingProps,
         listener: httpListener,
-        targetGroups: [this.TargetGroup],
+        action: ListenerAction.forward([this.targetGroup]),
```

Comment:
Any reason for changing this. Does the new version of cdk removes targetGroups props..??

Reply:
```ts
/**
   * Add a TargetGroup to load balance to
   *
   * @deprecated Use configureAction instead
   */
  public addTargetGroup(targetGroup: IApplicationTargetGroup) {
    this.configureAction(ListenerAction.forward([targetGroup]));
  }

```